### DISCUSSION TITLE
Turning off fluid solver

### DISCUSSION THREAD

**Discussion by raphaelouillon** [CliMA/Oceananigans.jl#2636] [Computational science!] [answered]
Hi all,
I was just curious to hear if it was possible to turn off the fluid solver and impose the velocity field directly. I am interested in a really simple problem with a constant velocity u(x,y,z,t)= U and a tracer that is passively advected, so I don't need to solve for the velocity field.

**Comment by francispoulin** [comment, marked as answer]
Yes, there is! You can use `PrescribedVelocityField`.  This will freeze the velocity and only evolve the tracers.  There should be something in one of the tests but I'm not sure where.  If you have problems finding it let me know, and I can try and dig something up.

**Reply by raphaelouillon** [reply]
great, thanks!

**Comment by simone-silvestri** [comment]
you can find an example of usage here

https://github.com/CliMA/Oceananigans.jl/blob/main/validation/solid_body_rotation/solid_body_tracer_advection.jl

**Reply by raphaelouillon** [reply]
Thanks!
